### ISSUE TITLE
 'flanking_bps' content appear to be occasionally incorrect + dependent on the order of variants in input VCF file.

### ISSUE THREAD

**Issue by sripakpa** [mskcc/vcf2maf#77]
Hi @ckandoth,

I am using vcf2maf (v1.6.8) + VEP 84.

I noticed that 'flanking_bps' content appear to be occasionally incorrect + dependent on the order the variants appear the input VCF file.

Here are test data to hopefully reproduce the issue.
Test VCF #1: 
```
##fileformat=VCFv4.2
#CHROM                POS       ID      REF                        ALT  QUAL  FILTER  INFO
1                     27092739  123456  A                          AG   .     .       .
1                     27023211  123457  ACGCGGGCCCTAGGCCCG         A    .     .       .
1                     27023211  123458  ACGCGGGCCCTAGGCCCGCCC      A    .     .       .
1                     27023211  123459  ACGCGGGCCCTAGGCCCGCCCTGAA  ACT  .     .       .
```

Test VCF #2:
```
##fileformat=VCFv4.2
#CHROM                POS       ID      REF                        ALT  QUAL  FILTER  INFO
1                     27023211  123458  ACGCGGGCCCTAGGCCCGCCC      A    .     .       .
1                     27023211  123459  ACGCGGGCCCTAGGCCCGCCCTGAA  ACT  .     .       .
1                     27092739  123456  A                          AG   .     .       .
1                     27023211  123457  ACGCGGGCCCTAGGCCCG         A    .     .       .
```

Note that data in the two test VCF files are the same except that rows are shuffled. This shouldn't affect the output 'flanking_bps'. However for test data #1, all 3 variants at position '27023211' have flanking_bps = AACGCGGGCCCTAGGCCCGC**CCTGAAC**. For test data #2, all 3 variants at position '27023211' have flanking_bps = AACGCGGGCCCTAGGCCCGC.

**Comment by ckandoth** [comment]
Thanks for reporting this. Now fixed in b450b4d. I did not expect VCFs with multiallelic positions in separate lines. The hashmap that gathered flanking bps, was previously only indexing by `CHROM` and `POS`, which means bps for only the last `1:27023211` variant call would be retained, and reused for all 3 variant calls. The hashmap now indexes by `CHROM` and `POS+length(REF)`.

**Comment by sripakpa** [comment]
I see. Thank you for fixing this!
